apis/placement/v1alpha1/disruptionbudget_types.go

```diff
@@ -0,0 +1,149 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope=Cluster,categories={fleet,fleet-placement},shortName=crpdb
+// +kubebuilder:subresource:status
+// +kubebuilder:storageversion
+
+// ClusterResourcePlacementDisruptionBudget is the policy applied to a ClusterResourcePlacement
+// object that specifies its disruption budget, i.e., how many placements (clusters) can be
+// down at the same time due to **voluntary** disruptions (e.g., evictions).
+//
+// To apply a ClusterResourcePlacementDisruptionBudget to a ClusterResourcePlacement, use the
```

Comment:
is it possible to do a selector?

Reply:
Sure, I will update the definition 👌

Reply:
I actually like having DisruptionBudget as part of a CRP. It only has two fields. Reusing the two fields across CRPs -> fragmented experience for a user who only manages one CRP. Instead of having one CRP that contains disruptionbudget, now the user need to define two objects.

Making the two fields part of the CRP provides simpler experience instead of adding one more level of indirection.

Reply:
When you have DisruptionBudget with CRP selectors -> N CRPs, there will be M DisruptionBudget upon one CRP. Both implementation and readability (users need to think about which DisruptionBudget applies to a given CRP) become much more complicated.

Reply:
I think using a selector probably is needed for blue-green upgrades with two deployments for the same app: blue deployment, and green deployment. So you can say I want to have at least N instances running for blue+green.

However, in this API, we claimed that we have 1-1 link between CRP and PDB.

Reply:
Just to document a few pros and cons I've collected in the past days:

Inline:

* Pros: 1 DB : 1 CRP mapping that is easy to reason about + easy to track
* Cons: less flexible + too much stuff in the CRP API

Individual object:

* Pros: flexible (1 DB : N CRP mapping) + can have status for users to inspect current status (X disruptions allowed)
    * for K8s PDB, M : N mapping is not allowed; a 500 error will be returned if multiple PDBs references the same pod [link](https://kubernetes.io/docs/concepts/scheduling-eviction/api-eviction/)
* Cons: more difficult to reason about + more difficult to track